notebooks/compare_file_structures.ipynb

```diff
@@ -1 +1,1069 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# File structures comparison\n",
+    "\n",
+    "This notebook can be used to compare different formats of the input data, like verse tagging, how are the poems grouped into sequences, ..., by looking at the validation loss."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install -r  ../requirements.txt"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "W6y8saYIIMjj",
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "from datasets import load_dataset\n",
+    "from enum import auto, Enum\n",
+    "from functools import partial\n",
+    "from happytransformer import fine_tuning_util, HappyGeneration\n",
+    "from huggingface_hub import login, notebook_login\n",
+    "import math\n",
+    "import numpy as np\n",
+    "import os\n",
+    "import pandas as pd\n",
+    "from pathlib import Path\n",
+    "import re\n",
+    "import transformers\n",
+    "from transformers import default_data_collator, Trainer, TrainingArguments\n",
+    "from transformers.optimization import SchedulerType\n",
+    "from transformers.trainer_utils import get_last_checkpoint\n",
+    "import torch\n",
+    "import torch.nn.functional as F\n",
+    "from typing import List, Tuple"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "run_as_standalone_nb = True\n",
+    "\n",
+    "\n",
+    "if run_as_standalone_nb:\n",
+    "    import sys    \n",
+    "    root_lib_path = Path('AI').resolve()\n",
+    "    if not root_lib_path.exists():\n",
+    "        !git clone https://github.com/Poems-AI/AI.git\n",
+    "    if str(root_lib_path) not in sys.path:\n",
+    "        sys.path.insert(0, str(root_lib_path))\n",
+    "    \n",
+    "    !pip install happytransformer\n",
```

Comment:
I think we can move the `!pip install AI/requirements.txt` here and drop the specific installs I had added previously. So, only install the requirements when the user is executing the notebook alone, like in Colab or Kaggle.

Now that I added installation instructions to README.md, when `run_as_standalone_nb == False`, i.e. the user is executing the notebook from a copy of the project (inside its directory structure), should we assume the requirements are already installed? I'm not sure.